src/ServiceControl/Recoverability/Retrying/Infrastructure/ReturnToSender.cs

```diff
@@ -30,23 +37,51 @@ public virtual async Task HandleMessage(MessageContext message, IDispatchMessage
 
             if (outgoingHeaders.TryGetValue("ServiceControl.Retry.Attempt.MessageId", out var attemptMessageId))
             {
-                var result = await bodyStorage.TryFetch(attemptMessageId)
-                    .ConfigureAwait(false);
-                if (result.HasResult)
+                if (outgoingHeaders.Remove("ServiceControl.Retry.HasAttachment"))
                 {
-                    using (result.Stream)
+                    var result = await bodyStorage.TryFetch(attemptMessageId)
```

Comment:
It might be worth refactoring out each method here to get some clarity on what's going on